sys/vm/vm_fault.c

```diff
@@ -348,15 +367,10 @@ vm_fault_soft_fast(struct faultstate *fs)
 				fs->fault_type |= VM_PROT_WRITE;
 		}
 	}
-#endif
-#if __has_feature(capabilities)
-	if (fs->first_object->flags & OBJ_NOLOADTAGS)
-		fs->fault_type |= PMAP_ENTER_NOLOADTAGS;
-	if (fs->first_object->flags & OBJ_NOSTORETAGS)
-		fs->fault_type |= PMAP_ENTER_NOSTORETAGS;
 #endif
 	rv = pmap_enter(fs->map->pmap, vaddr, m_map, fs->prot, fs->fault_type |
-	    PMAP_ENTER_NOSLEEP | (fs->wired ? PMAP_ENTER_WIRED : 0), psind);
+	    PMAP_ENTER_NOSLEEP | (fs->wired ? PMAP_ENTER_WIRED : 0) |
+	    pmap_tag_flags(fs->first_object), psind);
```

Comment:
Just stylistic, but given that the flags argument is now spanning three lines with `|`, maybe a local variable wouldn't be remiss.

Reply:
That is the sort of change that caused mismerges with later changes in FreeBSD (having a local variable).  I was intentional about avoiding the local variable as I think this approach has a better chance of dealing with conflicts in upstream changes in the future.